mantidproject/mantid
# Move IPeaksWorkspace properties -> PeaksWorkspace properties

**Description of work.**

TL;DR: This makes all algorithms that take peaks workspaces use `PeaksWorkspace` and not `IPeaksWorkspace` and exports a new workspace property (`PeaksWorkspaceProperty`) to python.

---

There's an inconsistency with the code base where in some places we use workspace properties that use `IPeaksWorkspace` and some that use `PeaksWorkspace`. This normally doesn't matter too much until you try and chain algorithms together in python. 

Before this PR we didn't have a `PeaksWorkspaceProperty` so algorithms that use `PeaksWorkspace` as output properties could be marked as `StoreInADS=False` because there was no way to get the value from the property.

Adding this causes a bigger problem because now we have a mix of algorithms that have `IPeaksWorkspace` properties and ones which have `PeaksWorkspace` properties. Grepping the code base suggests that there are many more algorithms that require `PeaksWorkspace` compare to `IPeaksWorkspace` and some of the algorithms which do use `IPeaksWorkspace` actually downcast to a `PeaksWorkspace` anyway.


**To test:**
 - Check all unit & system tests are passing
 - Code review


Does this update require release notes?


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
